tests/HostnameValidatorTest.php

```diff
@@ -169,8 +148,8 @@ public function testInvalidDoubledIdn()
     public function testURI()
     {
         $valuesExpected = array(
-            array(true, array('localhost', 'example.com', '~ex%20ample')),
-            array(false, array('§bad', 'don?t.know', 'thisisaverylonghostnamewhichextendstwohundredfiftysixcharactersandthereforshouldnotbeallowedbythisvalidatorbecauserfc3986limitstheallowedcharacterstoalimitoftwohunderedfiftysixcharactersinsumbutifthistestwouldfailthenitshouldreturntruewhichthrowsanexceptionbytheunittest')),
+            array(true, array('localhost', 'example.com')),
+            array(false, array('~ex%20ample')),
```

Comment:
Why is this now invalid?

Reply:
This has to be valid according to rfc 2822 which allows certain special characters as first character. but FILTER_FLAG_HOSTNAME does not allow any special characters as the first character in the host name. So, I was a little bit conflicted to allow those special characters as host name.

Reply:
Let's discuss this in person. Looks like it could be obsolete, but I have to take a deeper look.

Reply:
test redundant as we are not supporting uri validation and dot-atom format.